.circleci/config.yml

```diff
@@ -385,6 +385,116 @@ jobs:
+            # fallback to using the latest cache if no exact match is found
+            - v1-dependencies-
+
+      - run:
+          name: install dependencies
+          command: |
+            cd doc
+            python3 -m venv venv
+            . venv/bin/activate
+            npm install electron@1.8.4
+            npm install orca
+            pip install -r requirements.txt
+            if [ "${CIRCLE_BRANCH}" == "master" ]; then
+              pip uninstall plotly
+              cd ../packages/python/plotly
+              python3 setup.py install
+              cd ../../../doc      
+            fi 
+            echo 'export PATH="/home/circleci/project/doc/node_modules/.bin:$PATH"' >> $BASH_ENV
+            sudo apt-get install -y poppler-utils libxtst6 xvfb libgtk2.0-0 libgconf-2-4 libnss3 libasound2
+            cd ..
+
+      - save_cache:
+          paths:
+            - ./doc/venv
+            - ./doc/node_modules
+          key: v1-dependencies-{{ checksum "doc/requirements.txt" }}
+
+      - run:
+          name: make html
+          command: |
+            cd doc
+            . venv/bin/activate
+            echo ${mapbox_token} > python/.mapbox_token
+            make -kj8 || make -kj8
+            curl https://raw.githubusercontent.com/plotly/documentation/source-design-merge/front-matter-ci.py > front-matter-ci.py
+            curl https://raw.githubusercontent.com/plotly/documentation/source-design-merge/check-or-enforce-order.py > check-or-enforce-order.py
+            python front-matter-ci.py build/html
+            python check-or-enforce-order.py build/html
+            if [ "${CIRCLE_BRANCH}" == "doc-prod" ]; then
+              cd build/html
+              git init
+              git config user.name nicolaskruchten
+              git config user.email nicolas@plot.ly
+              git add *
+              git commit -m build
+              git push --force git@github.com:plotly/plotly.py.git doc-prod:built
+              rm -rf .git
+              cd ../..
+            fi
+            tar -zcf build/html.tgz build/html
+            rm -rf build/html build/ipynb
+            cd ..
+
+      - run:
+          name: make doc
+          command: |
+            cd doc
+            . venv/bin/activate
+            cd apidoc
+            make html
+            if [ "${CIRCLE_BRANCH}" == "doc-prod" ]; then
+              cd _build/html
+              touch .nojekyll
+              git init
+              git config user.name emmanuelle
+              git config user.email emma@plot.ly
+              git add *
+              git add .nojekyll
+              git commit -m build
+              git push --force git@github.com:plotly/plotly.py-docs.git master:gh-pages
+              rm -rf .git
+              cd ../..
+            fi
+
+            cd ../..
+
+
+      - store_artifacts:
+          path: doc/build
+          destination: doc/build
+     
+ 
+      - store_artifacts:
```

Comment:
is this step really necessary? It adds like 45s to the build, which is already at 5min...

Reply:
We can remove it I think. It can be nice when reviewing a PR because then you don't have to do the whole process of getting the branch and building the doc to look at how changes are rendered, but it's more useful for the build of tutorials than the apidoc (for which changes won't be that frequent).